node/node.go

```diff
@@ -568,13 +646,28 @@ func (node *Node) Start() error {
 					return libp2p_pubsub.ValidationAccept
 
 				case proto.Node:
-					// TODO push the message parsing here, so can ban
+					// node message is almost empty
+					if len(openBox) <= p2pNodeMsgPrefixSize {
+						return libp2p_pubsub.ValidationReject
+					}
+					validMsg, actionType, err := node.validateNodeMessage(
+						context.TODO(), openBox,
+					)
+					if err != nil {
+						// TODO (lc): block peers sending error messages
+						errChan <- withError{err, msg.GetFrom()}
+						return libp2p_pubsub.ValidationReject
+					}
 					msg.ValidatorData = validated{
 						consensusBound: false,
 						handleE:        node.HandleNodeMessage,
-						handleEArg:     openBox,
+						handleEArg:     validMsg,
+						actionType:     actionType,
 					}
+					return libp2p_pubsub.ValidationAccept
 				default:
+					// ignore garbled messages
+					atomic.AddUint32(&node.NumIgnoredMessages, 1)
```

Comment:
Shall this message be rejected since it does not have valid message type?

Reply:
good point, will do.